### PR TITLE
Moving libcamera documentation

### PR DESCRIPTION
**Pull request by aallan**
Moving the `libcamera` documentation up tree so that there is a `software/` top-level category to go with the `hardware/` top level category. Think I found all the pointers from the rest of the camera documentation to the page. But that said, the whole camera documentation needs refactored. It's scattered everywhere, which isn't good. More fodder for #1911.

### CODE REVIEW

File: usage/camera/README.md [hunk 1]
```diff
@@ -29,9 +29,9 @@ Advanced features, along with some hints and tips, are described in the followin
 libcamera is a new Linux API for interfacing to cameras. Raspberry Pi have been involved with the development of libcamera and are now using this sophisticated system for new camera software. This means Raspberry Pi are moving away from the firmware-based camera image processing pipeline (ISP) to a more open system.
 
 - [Libcamera's main website](http://libcamera.org/).
-- [Raspberry Pi libcamera implementation](../../linux/software/libcamera/README.md).
-- [Tuning guide for the Raspberry Pi cameras and libcamera](../../linux/software/libcamera/rpi_SOFT_libcamera_1p0.pdf)
-- [Writing a kernel module to support a new camera or capture chip](../../linux/software/libcamera/csi-2-usage.md)
+- [Raspberry Pi libcamera implementation](../../software/libcamera/README.md).
+- [Tuning guide for the Raspberry Pi cameras and libcamera](../../software/libcamera/rpi_SOFT_libcamera_1p0.pdf)
```
Comment (lurch) [inline usage/camera/README.md:33]: I guess this should point at `..._1p2.pdf`

Comment (aallan) [reply usage/camera/README.md:33]: Yes.

Comment (aallan) [reply usage/camera/README.md]: Fixed.

Comment (lurch) [reply usage/camera/README.md]: Lol, sorry for the conflict!

Comment (aallan) [reply usage/camera/README.md]: Doh! 😆

**Comment by aallan** [comment]
@lurch Is this fixed up by your tooling in the new documentation repo, and can be deleted and closed?

**Comment by lurch** [comment]
Yep.

**Pull request closed without merging**